src/server/server.js

```diff
@@ -218,13 +219,36 @@ SimulationServer.prototype._streamAppHostHtml = function (request, response) {
                 return '<script src="' + scriptSource + '"></script>';
             }).join('');
 
-            // Note we replace "default-src 'self'" with "default-src 'self' ws:" (in Content Security Policy) so that
-            // websocket connections are allowed (this relies on a custom version of send that supports a 'transform' option).
+            // Note we add "connect-src 'self' ws:" and "img-src 'blob:' (in Content Security Policy) so that
+            // websocket connections are allowed and the camera plugin works (this relies on a custom version
+            // of send that supports a 'transform' option).
+            var metaTagRegex = /<\s*meta[^>]*>/g;
+            var cspRegex = /http-equiv\s*=\s*(['"])Content-Security-Policy\1/;
+            var cspContent = /(content\s*=\s*")([^"]*)"/;
+            var maxCspTagLength = 1024;
```

Comment:
Maybe this could be declared only once at module level,
```
/**
 * Some doc if we need to clarify something.
 * @const
 */
var MAX_CSP_TAG_LENGTH = 1024;
```